WordPress/Classes/Services/BlogService.m

```diff
@@ -159,9 +159,14 @@ - (void)syncBlogsForAccount:(WPAccount *)account
 {
     DDLogMethod();
 
+    NSManagedObjectID *accountID = account.objectID;
     id<AccountServiceRemote> remote = [self remoteForAccount:account];
     [remote getBlogsWithSuccess:^(NSArray *blogs) {
         [self.managedObjectContext performBlock:^{
+            WPAccount *account = (WPAccount *)[self.managedObjectContext existingObjectWithID:accountID error:nil];
+            if (!account) {
+                return;
```

Comment:
I don't know. The sync didn't really fail, it's just not relevant anymore since the account is gone. Calling `failure` could end up in showing an error to the user that's not really a problem. Calling `success` feels misleading since nothing really synced. Not calling anything could lead to something waiting for the sync to finish, at least in theory.

No good solution here. I went for the simplest change possible, since the alternative right now is crashing.